app/code/Magento/Eav/Model/Entity/Attribute/Frontend/AbstractFrontend.php

```diff
@@ -49,22 +49,22 @@
     /**
      * @param \Magento\Eav\Model\Entity\Attribute\Source\BooleanFactory $attrBooleanFactory
      * @param CacheInterface $cache
-     * @param StoreResolverInterface $storeResolver
+     * @param StoreManagerInterface $storeManager
      * @param array $cacheTags
      * @codeCoverageIgnore
      */
     public function __construct(
         \Magento\Eav\Model\Entity\Attribute\Source\BooleanFactory $attrBooleanFactory,
         CacheInterface $cache = null,
-        StoreResolverInterface $storeResolver = null,
+        StoreManagerInterface $storeManager = null,
```

Comment:
that's a backward incompatible change,
and we can't just remove existing dependency and add another one instead.

So, existing dependency should be @deprecated 
while new dependency added at the end of the list